mongoimport/mongoimport.go

```diff
@@ -154,17 +161,34 @@ func (imp *MongoImport) ValidateSettings(args []string) error {
 		}
 	}
 
+	// deprecated
+	if imp.IngestOptions.Upsert == true {
+		imp.IngestOptions.Mode = modeUpsert
+	}
+
 	if imp.IngestOptions.UpsertFields != "" {
-		imp.IngestOptions.Upsert = true
+		if imp.IngestOptions.Mode == "" {
+			imp.IngestOptions.Mode = modeUpsert
+		} else if imp.IngestOptions.Mode == modeInsert {
+			return fmt.Errorf("can not use --upsertFields with --mode=insert")
+		}
 		imp.upsertFields = strings.Split(imp.IngestOptions.UpsertFields, ",")
 		if err := validateFields(imp.upsertFields); err != nil {
 			return fmt.Errorf("invalid --upsertFields argument: %v", err)
 		}
-	} else if imp.IngestOptions.Upsert {
+	} else if imp.IngestOptions.Mode != modeInsert {
 		imp.upsertFields = []string{"_id"}
 	}
 
-	if imp.IngestOptions.Upsert {
+	if imp.IngestOptions.Mode == "" {
+		imp.IngestOptions.Mode = modeInsert
+	} else if !(imp.IngestOptions.Mode == modeInsert ||
+		imp.IngestOptions.Mode == modeUpsert ||
+		imp.IngestOptions.Mode == modeMerge) {
+		return fmt.Errorf("invalid --mode argument: %v", imp.IngestOptions.Mode)
```

Comment:
This condition is unreachable, because of the go-flags choice restriction, and can safely be removed.